src/data/set/lattice.lean

```diff
@@ -4,13 +4,45 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jeremy Avigad, Leonardo de Moura, Johannes Hölzl, Mario Carneiro
 -/
 import order.complete_boolean_algebra
-import data.sigma.basic
-import order.galois_connection
 import order.directed
+import order.galois_connection
+
+/-!
+# The set lattice
+
+This file provides usual set notation for unions and intersections, a `complete_lattice` instance
+for `set α`, and some more set constructions.
+
+## Main declarations
+
+* `set.Union`: Union of an indexed family of sets.
+* `set.Inter`: Intersection of an indexed family of sets.
+* `set.sInter`: **s**et **Inter**. Intersection of sets belonging to a set of sets.
+* `set.sUnion`: **s**et **Union**. Intersection of sets belonging to a set of sets.
```

Comment:
This isn't in this file, but in core Lean: https://leanprover-community.github.io/mathlib_docs/init/data/set.html#set.sUnion

Though I do think mentioning it here makes sense.